sdk/eventhub/Azure.Messaging.EventHubs/tests/Snippets/Sample10_MockingReadingEventsTests.cs

```diff
@@ -0,0 +1,292 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Messaging.EventHubs.Consumer;
+using Azure.Messaging.EventHubs.Primitives;
+using Azure.Messaging.EventHubs.Producer;
+using Moq;
+using NUnit.Framework;
+
+namespace Azure.Messaging.EventHubs.Tests.Snippets
+{
+    [TestFixture]
+    public class Sample10_MockingReadingEventsTests
+    {
+        /// <summary>
+        ///   Performs basic unit test validation of the contained snippet.
+        /// </summary>
+        ///
+        [Test]
+        public async Task MockingReadEventsFromPartition()
+        {
+            #region Snippet:EventHubs_Sample10_MockingReadEventsFromPartition
+
+            var mockEventHubConsumerClient = new Mock<EventHubConsumerClient>();
+            var mockPartitionId = "sample partition id";
+
+            mockEventHubConsumerClient
+                .Setup(c => c.ReadEventsFromPartitionAsync(It.IsAny<string>(), It.IsAny<EventPosition>(), It.IsAny<CancellationToken>()))
+                .Returns(GetMockEventsFromPartition(mockPartitionId));
+
+            var consumer = mockEventHubConsumerClient.Object;
+            try
+            {
+                using CancellationTokenSource cancellationSource = new CancellationTokenSource();
+                cancellationSource.CancelAfter(TimeSpan.FromSeconds(30));
+
+                EventPosition startingPosition = EventPosition.Earliest;
+
+                await foreach (PartitionEvent partitionEvent in consumer.ReadEventsFromPartitionAsync(
+                    mockPartitionId,
+                    startingPosition,
+                    cancellationSource.Token))
+                {
+                    string readFromPartition = partitionEvent.Partition.PartitionId;
+                    ReadOnlyMemory<byte> eventBodyBytes = partitionEvent.Data.EventBody.ToMemory();
```

Comment:
just a thought do we need to add an assert or comment here, to point out the mocked data is being returned?

Reply:
Good call.   I'm comfortable with what you decide;  if it were me, I'd probably go the lazy route and just add a comment.